float32 x
float32 y
float32 r
bool was_found
